src/ext/java/org/opentripplanner/ext/gtfsgraphqlapi/generated/graphql-codegen.yml

```diff
@@ -59,6 +59,7 @@ config:
     Leg: org.opentripplanner.model.plan.Leg#Leg
     Mode: String
     TransitMode: org.opentripplanner.ext.gtfsgraphqlapi.generated.GraphQLTypes.GraphQLTransitMode#GraphQLTransitMode
+    OccupancyStatus: String
```

Comment:
Please map directly into the schema enum type rather than string. It makes it trivial to only return valid types in the implementation.

Reply:
I've thought about this but if we reuse the generated types here (like we already do in some cases) it creates a weird cyclical dependency of the code that generates types depending on generated types. I've continued using Strings as Hannes had originally used for enums but I've also thought it's a bit unsafe.

Reply:
I'm not sure I understand the problem. Lets discuss tomorrow.

Reply:
If we continue using strings, things like https://github.com/opentripplanner/OpenTripPlanner/pull/5368 keep happening.

Reply:
> I'm not sure I understand the problem. Lets discuss tomorrow.

This file defines the configuration for the codegen that generates both the GraphQLTypes and the GraphQLDataFetchers file. I don't think the codegen code is actually aware of if something exists or doesn't exist so I doubt something would break If we, for example, remove the Types file and try to run the codegen again.

Reply:
But wouldn't the file just be regenerated?

Reply:
It would but if the implementation of the codegen becomes "smarter" and checks that a mapper class actually exists while doing the codegen, this might break depending on the order it does things.

Reply:
Yes if you define an invalid class path here, it will still use it happily while generating the data fetchers. Lets discuss if this is actually a "circular dependency" or not in tomorrow's dev meeting.

Reply:
Then the compilation will fail.

Yes, lets discuss this tomorrow.

Reply:
We discussed today that generated types in the fetchers are a good thing.